fdbrpc/TokenSign.cpp

```diff
@@ -74,21 +77,18 @@ Standalone<SignedAuthTokenRef> signToken(AuthTokenRef token, StringRef keyName,
 	auto mdctxGuard = ScopeExit([mdctx]() { ::EVP_MD_CTX_free(mdctx); });
 	if (1 != ::EVP_DigestSignInit(mdctx, nullptr, ::EVP_sha256() /*Parameterize?*/, nullptr, key))
 		traceAndThrow("SignTokenInitFail");
-	if (1 != ::EVP_DigestSignUpdate(mdctx, tokenStr.begin(), tokenStr.size()))
+	if (1 != ::EVP_DigestSignUpdate(mdctx, str.begin(), str.size()))
 		traceAndThrow("SignTokenUpdateFail");
 	auto sigLen = size_t{};
 	if (1 != ::EVP_DigestSignFinal(mdctx, nullptr, &sigLen)) // assess the length first
 		traceAndThrow("SignTokenGetSigLenFail");
 	auto sigBuf = new (arena) uint8_t[sigLen];
 	if (1 != ::EVP_DigestSignFinal(mdctx, sigBuf, &sigLen))
 		traceAndThrow("SignTokenFinalizeFail");
-	ret.token = tokenStr;
-	ret.signature = StringRef(sigBuf, sigLen);
-	ret.keyName = StringRef(arena, keyName);
-	return ret;
+	return StringRef(sigBuf, sigLen);
 }
 
-bool verifyToken(SignedAuthTokenRef signedToken, StringRef publicKeyDer) {
+bool es256Verify(StringRef str, StringRef signature, StringRef publicKeyDer) {
```

Comment:
Code here isn't specific to ES256. It applies to PKI in general and only fixed part is using SHA256 as message digest ("S256" of "ES256")

Reply:
You're right -- fixing this.